File: src/main/java/no/nav/bidrag/beregn/underholdskostnad/beregning/Underholdskostnadberegning.java
package no.nav.bidrag.beregn.underholdskostnad.beregning;

import no.nav.bidrag.beregn.underholdskostnad.bo.BeregnUnderholdskostnadGrunnlagPeriodisert;
import no.nav.bidrag.beregn.underholdskostnad.bo.ResultatBeregning;

public interface Underholdskostnadberegning {

  ResultatBeregning beregn(
      BeregnUnderholdskostnadGrunnlagPeriodisert beregnUnderholdskostnadGrunnlagPeriodisert);

  Double beregnBarnetilsynMedStonad(
      BeregnUnderholdskostnadGrunnlagPeriodisert beregnUnderholdskostnadGrunnlagPeriodisert);

  static Underholdskostnadberegning getInstance(){
    return new UnderholdskostnadberegningImpl();
  }


}

File: src/main/kotlin/no/nav/bidrag/beregn/underholdskostnad/beregning/UnderholdskostnadBeregning.kt
package no.nav.bidrag.beregn.underholdskostnad.beregning

import no.nav.bidrag.beregn.underholdskostnad.bo.GrunnlagBeregning
import no.nav.bidrag.beregn.underholdskostnad.bo.ResultatBeregning


fun interface UnderholdskostnadBeregning {
    fun beregn(grunnlag: GrunnlagBeregning, barnetrygdIndikator: String): ResultatBeregning

    companion object {
        fun getInstance(): UnderholdskostnadBeregning {
            return UnderholdskostnadBeregningImpl()
        }
    }
}

